sklearn/preprocessing/tests/test_discretization.py

```diff
@@ -215,16 +230,76 @@ def test_encode_options():
                        .fit_transform(Xt_1).toarray(),
                        Xt_3.toarray())
     assert_raise_message(ValueError, "inverse_transform only supports "
-                         "'encode = ordinal'. Got 'encode = onehot' "
+                         "'encode = ordinal'. Got encode='onehot' "
                          "instead.", est.inverse_transform, Xt_2)
 
 
 def test_one_hot_encode_with_ignored_features():
     est = KBinsDiscretizer(n_bins=3, ignored_features=[1, 2],
-                           encode='onehot-dense').fit(X)
+                           encode='onehot-dense', strategy='uniform').fit(X)
     Xt = est.transform(X)
     Xt_expected = [[1, 0, 0, 1, 0, 0, 1.5, -4],
                    [0, 1, 0, 1, 0, 0, 2.5, -3],
                    [0, 0, 1, 0, 1, 0, 3.5, -2],
                    [0, 0, 1, 0, 0, 1, 4.5, -1]]
     assert_array_equal(Xt_expected, Xt)
+
+
+def test_invalid_strategy_option():
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3], strategy='invalid-strategy')
+    assert_raise_message(ValueError, "Valid options for 'strategy' are "
+                         "('uniform', 'quantile', 'kmeans'). "
+                         "Got strategy='invalid-strategy' instead.",
+                         est.fit, X)
+
+
+@pytest.mark.parametrize(
+    'strategy, expected_2bins, expected_3bins',
+    [('uniform', [0, 0, 0, 0, 1, 1], [0, 0, 0, 0, 2, 2]),
+     ('kmeans', [0, 0, 0, 0, 1, 1], [0, 1, 1, 1, 2, 2]),
+     ('quantile', [0, 0, 0, 1, 1, 1], [0, 0, 1, 1, 2, 2])])
+def test_nonuniform_strategies(strategy, expected_2bins, expected_3bins):
+    X = np.array([0, 1, 2, 3, 9, 10]).reshape(-1, 1)
+
+    # with 2 bins
+    est = KBinsDiscretizer(n_bins=2, strategy=strategy, encode='ordinal')
+    Xt = est.fit_transform(X)
+    assert_array_equal(expected_2bins, Xt.ravel())
+
+    # with 3 bins
+    est = KBinsDiscretizer(n_bins=3, strategy=strategy, encode='ordinal')
+    Xt = est.fit_transform(X)
+    assert_array_equal(expected_3bins, Xt.ravel())
+
+
+def test_kmeans_strategy():
```

Comment:
Could you explain the purpose of the test (maybe a comment)? It seems a bit strange (though I understand how it pass) and honestly seems redundant from my side.

Reply:
It is indeed redundant so I removed it.